### PR TITLE
adding scriptId property

### PR DESCRIPTION
**Pull request by thlorenz**
as discussed in #1

### CODE REVIEW

**Comment by thlorenz** [comment]
Just realized that the scriptId is not what I thought it was (the function id).

Instead it's the id of the file in which the function was defined. So that's still useful, but maybe not as much as I thought.

However there must be a way to get a function's id as well. So we should keep that in mind to add in the future.

Pulling @bmeck in here since he may know something.

**Comment by thlorenz** [comment]
I found `SnapshotId` that shows up in a heapdump, but looks like that concept only exists for a HeapGraphNode https://github.com/thlorenz/io.js/blob/v1.x/deps/v8/include/v8-profiler.h#L272

@chrisdickinson suggested to look into [GetIdentityHash](https://github.com/thlorenz/io.js/blob/v1.x/deps/v8/src/objects.h#L1657) which looks interesting.

We'd need to investigate though how this relates to `SnapshotId`, otherwise it may not be quite that useful for cross referencing this with snapshots.

**Comment by thlorenz** [comment]
Also realized that under node `0.10` `ScriptId` and `GetDisplayName` don't exist, so this would break compatibility.

Would you be ok with only adding those only for `0.12` and greater?
Makes the API inconsistent WRT to return value, but at least we'd get the best possible in either version.

**Comment by thlorenz** [comment]
Updated to maintain `0.10` compatibility. `scriptId` is not set when `<v0.12`.

**Comment by vkurchatkin** [comment]
> Would you be ok with only adding those only for 0.12 and greater?

I'm ok with this, but could you also test these conditionally? Right now test passes on 0.10 which indicates that `isNaN` is not ideal to test this

File: README.md [hunk 1]
```diff
@@ -18,7 +18,8 @@ var origin = FunctionOrigin(someFn);
 
  - `file`;
  - `line`;
- - `column`.
+ - `column`
+ - `scriptId` (`>=v0l.12` only).
```
Comment (vkurchatkin) [inline README.md:22]: What's `l`?

**Comment by vkurchatkin** [comment]
> We'd need to investigate though how this relates to SnapshotId, otherwise it may not be quite that useful for cross referencing this with snapshots.

Let's defer then. Don't want to throw random integers in, that we don't know what are good for :-)

**Comment by bmeck** [comment]
it is separate from snapshot id, and unrelated technically

File: README.md (hunk 1 above)
Comment (thlorenz) [inline README.md:22]: Sorry typo.

**Pull request closed without merging**